Sprint_07

BezrukovAndrej opened the pull request.

BezrukovAndrej committed:
согласно спринту 7

protos17 commented on MovieQuizTests/MoviesLoaderTests.swift:
```diff
@@ -0,0 +1,94 @@
+        loader.loadMovies { result in
+            switch result {
+            case .success(let movies):
+                XCTAssertEqual(movies.items.count, 2)
```
Тест не проходит, потому что здесь пропустил expectation.fulfill()

protos17 commented on MovieQuizTests/MoviesLoaderTests.swift:
```diff
@@ -0,0 +1,94 @@
+import XCTest
+@testable import MovieQuiz
+
+class MoviesLoaderTests: XCTestCase {
```
Класс необходимо пометить как final

protos17 commented on MovieQuizTests/ArrayTests.swift:
```diff
@@ -0,0 +1,24 @@
+        let value = array[safe: 2]
+        
+        XCTAssertNotNil(value)
+        XCTAssertEqual(value, 2)
```
Тест не проходит, потому что ты берешь элемент по индексу 2 и сравниваешь со значением 2, в то время как элемент в массиве по индексу 2 - "3"

protos17 commented on MovieQuizTests/ArrayTests.swift:
```diff
@@ -0,0 +1,24 @@
+        
+        let value = array[safe: 20]
+        
+        XCTAssertNotNil(value)
```
Тест не проходит, потому что тут должен быть в ответе nil, а ты указал, что значение должно быть не nil - XCTAssertNil(value)

protos17 commented on MovieQuizUITests/MovieQuizUITests.swift:
```diff
@@ -0,0 +1,89 @@
+        
+        
+        let alert = app.alerts.firstMatch
+        alert.buttons.firstMatch.tap()
```
Тест не проходит, потому что тут нужно подождать после нажатия на кнопку алерта, прежде чем искать лейбл, добавь sleep(2)

protos17 commented on MovieQuizUITests/MovieQuizUITests.swift:
```diff
@@ -0,0 +1,89 @@
+    
+    func testYesButton() {
+        sleep(3)
+        let firstPoster = app.images["Poster"]
```
Тест не проходит, потому что первый скриншот нужно делать здесь, когда находишь первый постер let firstPosterData = firstPoster.screenshot().pngRepresentation

protos17 commented on MovieQuizUITests/MovieQuizUITests.swift:
```diff
@@ -0,0 +1,89 @@
+        
+        let indexLabel = app.staticTexts["Index"]
+        
+        XCTAssertEqual(fistPosterData, secondPosterData)
```
Здесь тест проходит, но он написан неверно, здесь два постера не должны быть одинаковыми, для этого как в комментарии выше, нужно скриншот делать сразу после того, как найдешь первый постер

protos17 commented on MovieQuizUITests/MovieQuizUITests.swift:
```diff
@@ -0,0 +1,89 @@
+        
+        XCTAssertTrue(alert.exists)
+        XCTAssertTrue(alert.label == "Этот раунд окончен!")
+        XCTAssertTrue(alert.buttons.firstMatch.label == "Сыграть ещё раз")
```
По своему опыт скажу, что здесь в лейблах логичнее использовать сравнение через XCTAssertEqual()

protos17 commented on MovieQuiz/Presentation/MovieQuizPresenter.swift:
```diff
@@ -0,0 +1,136 @@
+    private weak var viewController: MovieQuizViewControllerProtocol?
+    
+    
+    init(viewController: MovieQuizViewControllerProtocol = MovieQuizViewController()) {
```
Советую поудалять лишние отступы, материал по код стайлу оставил в результатах ревью

protos17 commented on MovieQuiz/Presentation/MovieQuizPresenter.swift:
```diff
@@ -0,0 +1,136 @@
+            return
+        }
+        
+        let givenAnswer = isYes
```
Сделал неплохо, но можно улучшить тем, что здесь нет необходимости вводить лишнюю константу, можешь сразу использовать isYes ниже

protos17 commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,171 +1,98 @@
+    func showAlert(){
+        let text = presenter.makeResultsMessage()
+        alertPresenter = AlertPresenter(modelToShowAlert: AlertModel.init(
+            title: "Этот раунд окончен!",
```
Советую все строки типа string выносить в отдельные константы, в большинстве приложений используют локализацию, поэтому такая привычка пригодится в дальнейшем

BezrukovAndrej committed:
согл. зам.

Подскажи в чем прикол:
testAlertDismiss() - с 6 раза , с testGameFinish() - с 4 раза  только проходят тесты  ?

BezrukovAndrej committed:
.

protos17 commented on MovieQuizUITests/MovieQuizUITests.swift:
```diff
@@ -0,0 +1,87 @@
+    func testAlertDismiss() {
+        while app.staticTexts["Index"].label != "10/10" {
+            app.buttons["No"].tap()
+        }
```
Здесь тоже из-за блокирования кнопок работает некорректно тест

protos17 commented on MovieQuizUITests/MovieQuizUITests.swift:
```diff
@@ -0,0 +1,87 @@
+    func testGameFinish() {
+        while app.staticTexts["Index"].label != "10/10" {
+            app.buttons["No"].tap()
+        }
```
Тест не работает по причине блокирования кнопок до показа следующего постера (1 секунда), чтобы это решить используй        
 for _ in 0..<10 {
       app.buttons["No"].tap()
       sleep(1)
  }

BezrukovAndrej committed:
.

BezrukovAndrej committed:
.

BezrukovAndrej committed:
..

BezrukovAndrej committed:
отдельный метод под кнопки
исправила по 2 тестам.
